### PR TITLE
[#326] [Part 2/2] Add Kover to sample projects

### LINKED ISSUE #326: Update code coverage tool to Kover instead of Jacoco
## Why

Jacoco doesn't work correctly when [updating to the latest versions of Gradle & Java](https://github.com/nimblehq/android-templates/pull/323)

[Kover](https://github.com/Kotlin/kotlinx-kover) will not only fix this, but it's also a lot more convenient to set up and to work with (good for the future)

## Who Benefits?

Developers, this is essential to provide a good developer experience

### PR DESCRIPTION
**Pull request by ryan-conway**
closes #326

### Part 1
- [x] Added Kover to the template
- [x] Updated github workflows
- [x] Remove Jacoco references from `buildSrc`

### Part 2
- [x] Remove Jacoco references from `sample-xml`
- [x] Remove Jacoco references from `sample-compose`

## Proof Of Work 📹

Sample XML
image

Sample Compose
Screenshot 2565-11-10 at 10 32 11

### CODE REVIEW

**Comment by Tuubz** [comment]
@ryan-conway A separate pull request would be super nice 🙇

File: sample-xml/app/build.gradle.kts [hunk 1]
```diff
@@ -55,14 +55,6 @@ android {
...
             signingConfig = signingConfigs[BuildType.DEBUG]
             buildConfigField("String", "BASE_API_URL", "\"https://jsonplaceholder.typicode.com/\"")
-            /**
-             * From AGP 4.2.0, Jacoco generates the report incorrectly, and the report is missing
-             * some code coverage from module. On the new version of Gradle, they introduce a new
-             * flag [testCoverageEnabled], we must enable this flag if using Jacoco to capture
-             * coverage and creates a report in the build directory.
-             * Reference: https://developer.android.com/reference/tools/gradle-api/7.1/com/android/build/api/dsl/BuildType#istestcoverageenabled
-             */
-            isTestCoverageEnabled = true
```
Comment (Tuubz) [inline sample-xml/app/build.gradle.kts:65]: We forgot to remove this inside the `data` module 🙏

**Review by Tuubz — APPROVED** [review]
Rest LGTM 🚀 Would be nice if we can get that bug regarding the test in `sample-xml` merged first, before merging this

**Review by doannimble — APPROVED** [review]

**Comment by Tuubz** [comment]
@ryan-conway I've created a PR to fix the bug: https://github.com/nimblehq/android-templates/pull/348 🚀

**Comment by Tuubz** [comment]
@ryan-conway The bug has been merged to `develop`, could we rebase and update the description + PoW? 🙇

**Review by chornerman — APPROVED** [review]
LGTM 🚀

**Pull request merged**